pkg/node/volumemgr.go

```diff
@@ -301,7 +301,8 @@ func (m *VolumeManager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 			// drive must be present in the system
 			drive, _ := m.crHelper.GetDriveCRByVolume(volume)
 			if drive != nil {
-				m.addVolumeStatusAnnotation(drive, volume.Name, apiV1.Removed)
+				delete(drive.Annotations,
```

Comment:
It's better to use special methods for annotations:

```golang
annotations := obj.GetAnnotations()
delete(annotations, annotation)
obj.SetAnnotations(annotations)
// Update obj
```

Reply:
func (meta *ObjectMeta) GetAnnotations() map[string]string            { return meta.Annotations }
(meta *ObjectMeta) SetAnnotations(annotations map[string]string) { meta.Annotations = annotations }
=)
But I should to agree that your suggestion is more elegant.